pageserver/src/tenant/timeline/timeline_wrappers.rs

```diff
@@ -103,12 +105,13 @@ impl TimelineRef {
 /// A wrapper around `Arc<Timeline>` that prevents it from being dropped fully and allows accessing timeline's methods.
 /// Not intended to be cloned or stored somewhere: use it to access timeline's properties and drop it afterwards.
 /// By reacquiring the guard periodically, the timeline-accessing processes can stop when the timeline is being shut down.
-pub struct TimelineGuard {
+pub struct TimelineGuard<'a> {
     timeline: Arc<Timeline>,
+    _phantom: PhantomData<&'a Timeline>,
```

Comment:
```suggestion
    _phantom: PhantomData<&'a ()>,
```

Small nit, I think this is more "customary".